JS/main.js

```diff
@@ -0,0 +1,194 @@
+    columnThree: [cellTwo, cellFive, cellEight], // column
+    forwardSlash: [cellZero, cellFour, cellEight], // diagonal
+    backSlash: [cellSix, cellFour, cellTwo]  // diagonal
+};
+
+/* --------------------- Button Event Listeners----------------------- */
+
+onePlayerStart.addEventListener("click", () => {
+    onePlayerStart.disabled = true;
+    twoPlayerStart.disabled = true;
+    whosUp.textContent = currentPlayer;
+    start();
+})
+
+twoPlayerStart.addEventListener("click", () => {
+    onePlayerStart.disabled = true;
+    twoPlayerStart.disabled = true;
+    start();
+})
+
+submit.addEventListener('click', function (event) {
+    if (playerOne !== "") {
+        namePlayerOne(event)
+    } else {
+        namePlayerTwo(event)
+    }
+    event.preventDefault();
+    text = document.getElementById('playerName').value;
+})
+
+newGame.addEventListener("click", clicky);
+
+// This should put the current player's click into the proper cell
+function useClickedCell(event) {
+    
+    if (event.target.textContent === "") {
+        if (currentPlayer === playerOne) {
+            event.target.textContent = "X";
+            clickedCells.push(event.target);
+            currentPlayer = playerTwo;
+            previousPlayer = playerOne;
+        } else if (currentPlayer === playerTwo) {
+            event.target.textContent = "O";
+            clickedCells.push(event.target);
+            currentPlayer = playerOne;
+            previousPlayer = playerTwo;
+        }
+    }
+    announceWinner();
+}
+
+//This will announce the winner when a win condition is met
+function announceWinner() {
+    console.log("Inside announceWinner!")
+    for (let combination of Object.values(winConditions)) {
+        if (combination[0].textContent === "") {
+        } else if (
+            combination[0].textContent === combination[1].textContent &&
+            combination[0].textContent === combination[2].textContent
+        ) {
+            winner = true;
+            provide.textContent = previousPlayer + " has WON!";
+            
+        }
+    }   restartGame();
+    //isTied();
+
+}
+
+// This will (eventually) allow the player name entered to be tracked
+
+function namePlayerOne(event) {
+    if (makePlayerName.value === "") {
+        playerOne = "X";
+        makePlayerName = "";
+    } else {
+        playerOne = makePlayerName.value;
+        makePlayerName = "";
+    }
+    console.log(playerOne + " is player one");
+
+}
+
+function namePlayerTwo(event) {
+    if (makePlayerName.value === "") {
+        playerTwo = "O";
+        makePlayerName = "";
+    } else {
+        playerTwo = makePlayerName.value;
+        makePlayerName = "";
+    }
+    console.log(playerTwo + " is player two");
+}
+
+// Provides ability to select the desired game (like in Guess the Number)
+
+function start() {
+    cellArray.forEach(function (clicky) {
+        clicky.addEventListener("click", useClickedCell);
+    });
```

Comment:
The way you are using your `forEach` is causing errors and you never define what `clicky` is